src/core/components/scroll-to-top.jsx

```diff
@@ -0,0 +1,85 @@
+import React, { useState, useEffect } from "react"
+
+const ScrollToTopButton = () => {
+  const [isVisible, setIsVisible] = useState(false)
+  const [hideTimeout, setHideTimeout] = useState(null) 
+
+  const styles = {
+    button: {
+      position: "fixed",
+      bottom: "20px",
+      right: "20px",
+      padding: "10px 20px",
+      fontSize: "18px",
+      backgroundColor: "rgba(125, 132, 146, 0.8)",
+      color: "#fff",
+      border: "2px solid rgba(125, 132, 146, 0.8)",
+      borderRadius: "5px",
+      cursor: "pointer",
+      boxShadow: "0px 4px 6px rgba(0, 0, 0, 0.1), 0 0 15px rgba(125, 132, 146, 0.6)",
+      transition: "opacity 0.5s ease, box-shadow 0.3s ease",
+      opacity: isVisible ? 1 : 0,
+      pointerEvents: isVisible ? "auto" : "none",
+    },
+    icon: {
+      width: "20px",
+      height: "20px",
+      fill: "#fff",
+    },
+  }
+
+  const toggleVisibility = () => {
```

Comment:
The toggleVisibility function handles both the visibility toggle and timeout reset, which could make the code harder to follow. Consider breaking it up for simplicity: keep the scroll visibility logic and timeout logic separate. Instead of adding this logic directly to that file, consider creating a custom hook. This approach would improve readability and maintain cleaner code.